File: typescript/template/CharDisplay.ts
import { AbstractDisplay } from './AbstractDisplay.ts';

export class CharDisplay extends AbstractDisplay {
  private char: string;

  constructor(char: string) {
    super();
    if (char.length === 1) {
      this.char = char;
    } else {
      throw new Error('Argument should be 1 character.');
    }
  }

  public open = (): string => {
    return '<';
  };

  public print = (_: boolean): string => {
    return this.char;
  };

  public close = (): string => {
    return '>';
  };
}

File: go/template/chardisplay/chardisplay.go
package chardisplay

import (
	"github.com/hitsuji-haneta/design-pattern/go/template/display"
)

type charDisplay struct {
	*display.AbstractDisplay
	char rune
}

func New(char rune) *charDisplay {
	cd := &charDisplay{&display.AbstractDisplay{}, char}
	cd.Printer = cd
	return cd
}

func (cd *charDisplay) Open() string {
	return "<"
}

func (cd *charDisplay) Print(flag bool) string {
	return string(cd.char)
}

func (cd *charDisplay) Close() string {
	return ">"
}